pydantic/fields.py

```diff
@@ -321,7 +321,7 @@ class MyModel(pydantic.BaseModel):
                 new_field_info.metadata = metadata
                 return new_field_info
 
-        return FieldInfo(annotation=annotation, frozen=final or None)
+        return FieldInfo(annotation=annotation, frozen=final or None)  # pyright: ignore[reportArgumentType]
```

Comment:
What's the issue with the argument type here?

Reply:
`annotation` is inferred as `type[Any] | type[Final]` because of the `if annotation is not typing_extensions.Final:` check on line 300. And with the recent changes in pyright regarding special forms (such as `Final`), `type[Final]` is not compatible with `type[Any]`. Hopefully this will also be resolved with the TypeForm PEP.